gitlab.go

```diff
@@ -123,7 +123,10 @@ func NewClient(httpClient *http.Client, token string) *Client {
 	}
 
 	c := &Client{client: httpClient, token: token, UserAgent: userAgent}
-	c.SetBaseURL(defaultBaseURL)
+	if err := c.SetBaseURL(defaultBaseURL); err != nil {
+		// should never happen since defaultBaseURL is our constant
+		panic(err)
+	}
```

Comment:
Hmm...  Like you say in the comment, `defaultBaseURL` is a constant we defined ourselves so this should never give an error.

On the other hand it's of course a good habit to check error values that are returned. But if we decide to do that here as well, then I would prefer to just return the error instead of throwing a panic. It feels a little wrong and dirty to have a library package throw a panic. If it just returns an error, the program using this package can decide how to handle it instead.

That being said, I do wonder what the added value is for checking the error in this particular case. Maybe the first test case we should add in this package is a test for this function. That way the test case can make sure this function works when code in the package is changed, which would make this error check here even more useless.

Thoughts?